L/libjulia_jll/Compat.toml

```
@@ -17,6 +17,8 @@ MPFR_jll = "4.0.2"
 libLLVM_jll = "8.0.1"
 
 ["1.5-1"]
-MPFR_jll = "4.1.0"
 OpenBLAS_jll = "0.3.9"
 libLLVM_jll = "9.0.1"
+
+["1.5-1.5.1"]
+MPFR_jll = "4.1.0"
```

I bet this is the problem

---

I think it should be
```suggestion
MPFR_jll = "=4.1.0"
```
or whatever is the right syntax in the registry

---

Hmm, why? So in libjulia_jll 1.5.1, we had a bad dependency on MPFR_jll 4.1.0; but that's fixed in the new libjulia_jll 1.5.3, which drops it?

---

Ah, but https://github.com/JuliaBinaryWrappers/SuiteSparse_jll.jl/releases/tag/SuiteSparse-v5.4.0%2B10 does indeed provide `SuiteSparse.v5.4.0.aarch64-apple-darwin.tar.gz`

So at least that (and possibly more) are shipping platforms not (yet) supported in Julia 1.5, it seems?

But I still don't understand where that 401 error comes from?

---

No, `MPFR_jll` is at version v4.1.0+1, which looks ok, the error must be from somewhere else.

> But I still don't understand where that 401 error comes from?

PkgServer?

---

So I manually tried this, and it seems to work fine for me in Julia 1.5.3, locally:
```
(proj-1.5) pkg> add https://github.com/JuliaBinaryWrappers/libjulia_jll.jl#main
   Updating git-repo `https://github.com/JuliaBinaryWrappers/libjulia_jll.jl`
   Updating registry at `~/.julia/registries/General`
   Updating git-repo `https://github.com/JuliaRegistries/General.git`
  Resolving package versions...
  Installed XZ_jll ──────── v5.2.4+3
  Installed LibUnwind_jll ─ v1.3.1+1
Downloading artifact: libjulia
Downloading artifact: XZ
...
julia> using libjulia_jll
[ Info: Precompiling libjulia_jll [5ad3ddd2-0711-543a-b040-befd59781bbf]
┌ Warning: Platform `aarch64-linux-gnu-cxx03-julia_version+1.5` is not an officially supported platform
└ @ Pkg.BinaryPlatforms /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/Pkg/src/BinaryPlatforms.jl:513
┌ Warning: Platform `aarch64-linux-gnu-cxx11-julia_version+1.5` is not an officially supported platform
...
┌ Warning: Platform `x86_64-w64-mingw32-cxx03-julia_version+1.5` is not an officially supported platform
└ @ Pkg.BinaryPlatforms /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/Pkg/src/BinaryPlatforms.jl:513
┌ Warning: Platform `x86_64-w64-mingw32-cxx11-julia_version+1.5` is not an officially supported platform
└ @ Pkg.BinaryPlatforms /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/Pkg/src/BinaryPlatforms.jl:513

julia>
```

So I am even more confused as to what's the problem here -- I thought the tests here were using Julia 1.5?

I can't really install it in Julia 1.6, though:
```
(.ci) pkg> add https://github.com/JuliaBinaryWrappers/libjulia_jll.jl#main
     Cloning git-repo `https://github.com/JuliaBinaryWrappers/libjulia_jll.jl`
    Updating git-repo `https://github.com/JuliaBinaryWrappers/libjulia_jll.jl`
    Updating registry at `~/.julia/registries/General`
    Updating git-repo `https://github.com/JuliaRegistries/General.git`
   Resolving package versions...
ERROR: Unsatisfiable requirements detected for package SuiteSparse_jll [bea87d4a]:
 SuiteSparse_jll [bea87d4a] log:
 ├─possible versions are: 5.4.0-5.4.1 or uninstalled
 ├─restricted to versions 5.4.0 by libjulia_jll [5ad3ddd2], leaving only versions 5.4.0
 │ └─libjulia_jll [5ad3ddd2] log:
 │   ├─possible versions are: 1.5.3 or uninstalled
 │   └─libjulia_jll [5ad3ddd2] is fixed to version 1.5.3+0
 └─restricted by julia compatibility requirements to versions: 5.4.1 or uninstalled — no versions left
```
But I *think* that doesn't matter for using it on Yggdrasil?

---

SuiteSparse should not cause problems as the 5.4.0+10 build doesn't exist in the registry and is not used according to the build log.

But LibUV_jll 2.0.0+11 is pulled in and contains `aarch64-apple-darwin`: [jll](https://github.com/JuliaBinaryWrappers/LibUV_jll.jl/commit/70dacb47f5), [compat](https://github.com/JuliaRegistries/General/blob/master/L/LibUV_jll/Compat.toml) and [versions](https://github.com/JuliaRegistries/General/blob/master/L/LibUV_jll/Versions.toml)

---

> and it seems to work fine for me in Julia 1.5.3

Because v1.5.2+ knows how to deal with the new architectures, v1.5.1 (used here in CI) doesn't.

> But I think that doesn't matter for using it on Yggdrasil?

Sure, but this is a symptom that there is a version that should be compatible only for v1.6+ that is slipping in the environment.  We want to fix that anyway

---

> But LibUV_jll 2.0.0+11 is pulled in and contains `aarch64-apple-darwin`

Good spot!  #25311